packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -678,6 +673,12 @@ class ReactDOMServerRenderer {
     context: Object,
     parentNamespace: string,
   ): string {
+    invariant(
```

Comment:
...but this is. Seems unfortunate to run this check every time.

(Also, if we do this I’d expect it to happen outside of `renderDOM` which renders host components only.)

Since we already read `.type` above when comparing to fragment maybe we should turn that branch into a switch?